File: scheduler/src/main/java/com/stats/aggregator/scheduler/tasks/HadoopTask.java
package com.stats.aggregator.scheduler.tasks;

import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Component;

@Component
public class HadoopTask {

    @Scheduled(cron = "0 10 * * * *")
    public void doEveryHour() {

        // task execution logic
    }

    @Scheduled(cron = "0 20 23 * * *")
    public void doEveryDay() {

        // task execution logic
    }

    @Scheduled(cron = "0 30 23 1 * *")
    public void doEveryMonth() {

        // task execution logic
    }

    @Scheduled(cron = "0 40 23 1 1 *")
    public void doEveryYear() {

        // task execution logic
    }
}

File: scheduler/src/main/scala/com/stats/aggregator/scheduler/tasks/HadoopTask.scala
package com.stats.aggregator.scheduler.tasks

import com.stats.aggregator.scheduler.services.contract.TRestApiClient
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Component

@Component class HadoopTask @Autowired()(restApiClient: TRestApiClient) {

  @Scheduled(initialDelay=10000, fixedRate=60000) def justForTest() {
    restApiClient getCurrentPrices "1"
    // task execution logic
  }

  @Scheduled(cron = "0 10 * * * *") def doEveryHour() {
    // task execution logic
  }

  @Scheduled(cron = "0 20 23 * * *") def doEveryDay() {
    // task execution logic
  }

  @Scheduled(cron = "0 30 23 1 * *") def doEveryMonth() {
    // task execution logic
  }

  @Scheduled(cron = "0 40 23 1 1 *") def doEveryYear() {
    // task execution logic
  }
}